runtime/onert/api/src/nnfw_api_internal.cc

```diff
@@ -1173,6 +1173,78 @@ NNFW_STATUS nnfw_session::set_backends_per_operation(const char *backend_setting
   return NNFW_STATUS_NO_ERROR;
 }
 
+NNFW_STATUS nnfw_session::train_set_traininfo(const nnfw_train_info *info)
+{
+  if (not isStateModelLoaded())
+  {
+    std::cerr << "Error during nnfw_session::train_set_traininfo : invalid state" << std::endl;
+    std::cerr << "nnfw_session::train_set_traininfo must be called in MODEL_LOADED state"
+              << std::endl;
+    return NNFW_STATUS_INVALID_STATE;
+  }
+
+  if (info == nullptr)
+  {
+    std::cerr << "nnfw_session::train_set_traininfo : info is nullptr" << std::endl;
+    return NNFW_STATUS_UNEXPECTED_NULL;
+  }
+
+  // after model loaded, it ensures that _train_info is not nullptr
+  assert(_train_info != nullptr);
+  assert(info != nullptr);
+
+  auto convertLossType = [](const int &type) {
+    if (type == NNFW_TRAIN_LOSS_MEAN_SQUARED_ERROR)
+      return onert::ir::train::LossCode::MeanSquaredError;
+    if (type == NNFW_TRAIN_LOSS_CATEGORICAL_CROSSENTROPY)
+      return onert::ir::train::LossCode::CategoricalCrossentropy;
+    else
+      throw std::runtime_error("not supported loss type");
+  };
+
+  auto convertLossReductionType = [](const int &type) {
+    if (type == NNFW_TRAIN_LOSS_REDUCTION_AUTO)
+      return onert::ir::train::LossReductionType::Invalid;
```

Comment:
```suggestion
      return onert::ir::train::LossReductionType::Auto;
```
Maybe it should be Auto type?

Reply:
Thank you for catching :) 
Updated